Cargo.toml

```diff
@@ -15,8 +15,8 @@ vendored = ["native-tls/vendored"]
 
 [dependencies]
 bytes = "1"
-native-tls = "0.2.1"
-hyper = { version = "0.14.2", default-features = false, features = ["tcp", "client"] }
+native-tls = { version = "0.2.7", features = ["alpn"] }
+hyper = { version = "0.14.2", default-features = false, features = ["tcp", "client", "http1", "http2"] }
```

Comment:
This forces http1 and http2 back in. It's unclear to me how to proceed here. I believe we either have to always activate these or have to add new public features of hyper-tls, which if you forget to activate them, you are not getting any ALPN. Though we could make those features active by default.

Reply:
Do we need to enable them in this crate? As long as we don't *send* ALPN by default, the user can opt-in after enabling the features.

Reply:
Yeah we can definitely just not send ALPN by default. It could then either be a feature of this crate (which then probably should delegate it further) or be some kind of constructor / builder / configure method of the HttpsConnector (and hoping that the appropriate features are active for hyper). I'll leave that decision up to you before I proceed further.

Reply:
I was thinking about this issue and the easiest implementation that comes to mind would be to have something like this in hyper:

```rust
/// Returns `true` if the `http2` feature is enabled
pub const fn http2_feature_enabled() -> bool {
    cfg!(feature = "http2")
}
```

The advantage is that enabling http2 in hyper would also enable it automatically in the TLS libraries. The downside is that it wouldn't be possible to disable the ALPN feature in hyper-tls. Also I've never seen something like this done in other crates so maybe it's a bad pattern to do in Rust?